pkg/sql/schemachanger/scbuild/builder.go

```diff
@@ -671,6 +680,49 @@ func (b *Builder) addNode(dir scpb.Target_Direction, elem scpb.Element) {
 	})
 }
 
+// getLockedTableDescriptor returns a table descriptor guaranteed to be "locked"
+// (i.e., with no other concurrent schema changes), or else a
```

Comment:
The "locked" terminology comes from a (currently under-specified) concept of locking vs. non-locking changes to descriptors that @ajwerner and I have been discussing, where a locking change in a schema change corresponds roughly to a schema change that involves mutations and requires mutual exclusion. (Some long and not very digestible internal discussions: [1](https://cockroachlabs.slack.com/archives/CARM0PTK4/p1607110682076600), [2](https://cockroachlabs.slack.com/archives/DDQT4UW9K/p1614357598000200), [3](https://cockroachlabs.slack.com/archives/CARM0PTK4/p1613145290261200).)

I think the concept will need some work, but I'm inclined to keep the terminology for now. It is admittedly weird that the present implementation doesn't actually tracked "locked" descriptors in the builder at all, so it doesn't provide anything except possibly returning an error, because we assume all of our targets are for locking changes. I left some TODOs with some further thoughts on this.

I am open to changing the name of this method, though. Maybe `getTableDescriptorForLockingChange`?

Reply:
If the concept already exists then I'm cool with that. That wasn't obvious to me so I feel it would be worth defining that concept more explicitly in a comment somewhere, even if it's just:
```
// We define a _locking_ change in a schema change as corresponding roughly to a schema change that involves mutations and requires mutual exclusion.
```
I also like your idea of renaming the method like this because the current name implies that the descriptor is already locked.

Reply:
Renamed the method and added a comment.